easybuild/easyblocks/generic/mesonninja.py

```diff
@@ -53,12 +53,26 @@ def extra_options(extra_vars=None):
         extra_vars.update({
             'build_dir': [None, "build_dir to pass to meson", CUSTOM],
             'build_cmd': [DEFAULT_BUILD_CMD, "Build command to use", CUSTOM],
+            'build_type': [None, "Build type for meson, e.g. release."
+                                 "Replaces use of toolchain options debug, noopt, lowopt", CUSTOM],
+            'ndebug': [True, "Sets -Db_ndebug which in turn defines NDEBUG for C/C++ builds."
+                             "This disabled costly asserts in code, typical for production.", CUSTOM],
             'configure_cmd': [DEFAULT_CONFIGURE_CMD, "Configure command to use", CUSTOM],
             'install_cmd': [DEFAULT_INSTALL_CMD, "Install command to use", CUSTOM],
             'separate_build_dir': [True, "Perform build in a separate directory", CUSTOM],
         })
         return extra_vars
 
+    @property
+    def optimization(self):
+        """Optimization level"""
+        if self.toolchain.options.get('noopt', False):
+            return 0
+        elif self.toolchain.options.get('lowopt', False):
+            return 1
+        else:
+            return 2
+
```

Comment:
wait, what if we have `'opt': True` (as in, use `-O3`)?
surely this needs another elif?

Reply:
```suggestion
    @property
    def optimization(self):
        """Optimization level"""
        if self.toolchain.options.get('noopt', False):
            return 0
        elif self.toolchain.options.get('lowopt', False):
            return 1
        elif self.toolchain.options.get('opt', False):
            return 3
        else:
            return 2

```